sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -968,6 +968,11 @@ object DateTimeUtils {
    */
   def currentTimestamp(): Long = instantToMicros(Instant.now())
 
+  /**
+   * Obtains the current local date-time as microseconds since the epoch in the default time-zone.
+   */
+  def currentTimestampNTZ(): Long = localDateTimeToMicros(LocalDateTime.now())
```

Comment:
hmm, do you propose to return local date-time at JVM time zone?
```java
    /**
     * Obtains the current date-time from the system clock in the default time-zone.
     * <p>
     * This will query the {@link Clock#systemDefaultZone() system clock} in the default
     * time-zone to obtain the current date-time.
     * <p>
     * Using this method will prevent the ability to use an alternate clock for testing
     * because the clock is hard-coded.
     *
     * @return the current date-time using the system clock and default time-zone, not null
     */
    public static LocalDateTime now() {
        return now(Clock.systemDefaultZone());
    }
```
I would expect to get local date-time at the session time zone.

Reply:
The Java class `java.time.LocalDateTime` has a similar semantic to ANSI SQL timestamp without timezone type, and it is the most suitable to be an external type for `TimestampWithoutTZType`.

Reply:
@beliefer I am +1 with @MaxGekk , we should respect the conf "spark.sql.session.timeZone" here.

Reply:
Thanks. I got it.